source/retryable-reads/retryable-reads.rst

```diff
@@ -356,26 +360,43 @@ and reflects the flow described above.
     /* Allow ServerSelectionException to propagate to our caller, which
      * can then assume that no attempts were made. */
     server = selectServer();
-    connection = server.getConnection()   
+
+    /* PoolClearedException indicates the operation did not even attempt to
+     * create a connection, let alone execute the operation. This means we
+     * are always safe to attempt a retry. We do not need to update SDAM,
+     * since whatever error caused the pool to be cleared will do so itself. */
+    try {
+      connection = server.getConnection()
+    } catch (PoolClearedException poolClearedError) {
+      return executeRetry(command, session, poolClearedError);
```

Comment:
This unconditionally retries, which should be safe as per the comment in the code. That being said, I did not make this a requirement, as it could be hard to specify, at least for `PoolClosedError`s specifically. When we do the more general DRIVERS-746, I think this would be worth revisiting.